sdk/servicebus/azure-servicebus/azure/servicebus/_servicebus_sender.py

```diff
@@ -61,13 +68,15 @@ def _set_msg_timeout(self, timeout=None, last_exception=None):
         self._handler._msg_timeout = timeout * 1000  # type: ignore
 
     @classmethod
-    def _build_schedule_request(cls, schedule_time_utc, *messages):
+    def _build_schedule_request(cls, schedule_time_utc, send_span, *messages):
         request_body = {MGMT_REQUEST_MESSAGES: []}
         for message in messages:
             if not isinstance(message, ServiceBusMessage):
                 raise ValueError("Scheduling batch messages only supports iterables containing "
                                  "ServiceBusMessage Objects. Received instead: {}".format(message.__class__.__name__))
             message = transform_messages_to_sendable_if_needed(message)
+            trace_message(message, send_span)
```

Comment:
On when the send_span should be created and if it should be the parent of message...

Checked with @chradek, he pointed out the guideline from the [gist](https://gist.github.com/lmolkova/e4215c0f44a49ef824983382762e6b92#send-messages).
> SDK MUST set links before starting the send span. This enables sampling based on links sampling decisions.

Reply:
Good news here I believe, (and re: the above discussion), my understanding is that what the trace below shows aligns with what your comments would expect? (although to your point, yes, the message does get parented to the send, but this is as EH did)

```json
{
    "name": "ServiceBus.message",
    "context": {
        "trace_id": "0xa495af37c346711c49076f35a1e79c2e",
        "span_id": "0x5977498dc789253e",
        "trace_state": "{}"
    },
    "kind": "SpanKind.PRODUCER",
    "parent_id": "0xa50e404209d268d7",
    "start_time": "2020-11-18T18:42:03.174657Z",
    "end_time": "2020-11-18T18:42:03.174657Z",
    "status": {
        "canonical_code": "OK"
    },
    "attributes": {
        "az.namespace": "ServiceBus"
    },
    "events": [],
    "links": [],
    "resource": {
        "telemetry.sdk.language": "python",
        "telemetry.sdk.name": "opentelemetry",
        "telemetry.sdk.version": "0.13b0"
    }
}
{
    "name": "ServiceBus.send",
    "context": {
        "trace_id": "0xa495af37c346711c49076f35a1e79c2e",
        "span_id": "0xa50e404209d268d7",
        "trace_state": "{}"
    },
    "kind": "SpanKind.CLIENT",
    "parent_id": "0x4f9ec1e25cbf7536",
    "start_time": "2020-11-18T18:42:03.174657Z",
    "end_time": "2020-11-18T18:42:03.335652Z",
    "status": {
        "canonical_code": "OK"
    },
    "attributes": {
        "component": "servicebus",
        "az.namespace": "az.namespace",
        "message_bus.destination": "test",
        "peer.address": "<redacted>.servicebus.windows.net"
    },
    "events": [],
    "links": [
        {
            "context": {
                "trace_id": "0xa495af37c346711c49076f35a1e79c2e",
                "span_id": "0x5977498dc789253e",
                "trace_state": "{}"
            },
            "attributes": null
        }
    ],
    "resource": {
        "telemetry.sdk.language": "python",
        "telemetry.sdk.name": "opentelemetry",
        "telemetry.sdk.version": "0.13b0"
    }
}
```